service/history/shard/controller_test.go

```diff
@@ -112,6 +113,7 @@ func NewTestController(
 		ThrottledLogger:             resource.GetThrottledLogger(),
 		TimeSource:                  resource.GetTimeSource(),
 		TaskCategoryRegistry:        tasks.NewDefaultTaskCategoryRegistry(),
+		NewEventsCacheFn:            events.NewCacheFnProvider(),
```

Comment:
do we need to populate the new `EventsCache` field here as well?

Reply:
It is not used anywhere. Still do we need to add that?